Documentation/workflow/SystemProperties.md

```diff
@@ -157,13 +157,74 @@ categories:
     Enable logging for messages related to network activity.
   * `timing=bare`
     Enable logging of native code performance information, without
-    logging method execution timing information to a file.
+    logging method execution timing information to a file.  Timed
+    events are logged to `logcat` immediately, which affects the
+    accuracy of measurements and introduces an element of
+    unpredictability to measurements.
+  * `timing=fast-bare`
+    Similar to `timing=bare` above, but the timings aren't logged
+    until the `mono.android.app.DUMP_TIMING_DATA` broadcast is sent to
+    the application.  This can be done with `adb shell am broadcast -a
+    mono.android.app.DUMP_TIMING_DATA [PACKAGE_NAME]` command.
   * `timing`
     Enable logging of native code performance information, including
     method execution timing which is written to a file named
     `methods.txt`.  `timing=bare` should be used in preference to this
     category.
 
+#### Timing events format
+
+In any of the timing modes, the logged messages have the following
+format:
+
+```
+<TAG>: [<S>/<E>] <MESSAGE>; elapsed s:ms::ns
```

Comment:
What's the rationale for `<S>` and `<E>` being *numbers*, as opposed to some "more meaningful" descriptor?

If (when?) we add additional stages (perhaps not *that* plausible) or events (feels quite plausible), numbers will require updating, which in turn will mean that the output is dependent upon the version of xamarin-android being used, which feels…brittle.

Reply:
> What's the rationale for `<S>` and `<E>` being _numbers_, as opposed to some "more meaningful" descriptor?

 * Numbers don't require coming up with unique names, they are just monotonically increased.
 * The output is shorter, leaving more room for the actual message (logcat messages have a limit of 1023 characters)
 * The tag is meant for script/software parsing, humans reading the output need to pay attention only to the actual
   message, thus there's no need for the tag to have meaningful names

> 
> If (when?) we add additional stages (perhaps not _that_ plausible) or events (feels quite plausible), numbers will require updating, which in turn will mean that the output is dependent upon the version of xamarin-android being used, which feels…brittle.

The new events will just next numbers in sequence. This is no more brittle than adding a new event name, which also makes output dependent on the version...